rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,87 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
```

Comment:
Who's going to use the server health information if not the clients? A server itself doesn't care if it's unhealthy.

I'm not saying the transport package should be the source of the information, but it should "own" it in a sense, since how else will clients find out if they are connected to an unhealthy backend server? I picture the client and server packages as independent things both depending on the transport package (among others).